src/utils/astCreator.ts

```diff
@@ -91,10 +91,11 @@ export const blockArrowFunction = (
 export const functionExpression = (
   params: es.Pattern[],
   body: es.Statement[] | es.BlockStatement,
-  loc?: es.SourceLocation | null
+  loc?: es.SourceLocation | null,
+  id?: es.Identifier
 ): es.FunctionExpression => ({
   type: 'FunctionExpression',
-  id: null,
+  id: id || null,
```

Comment:
```suggestion
  id: id ?? null,
```

Minor nit: The [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) better captures the semantics here instead of `||` which is checking if `id` is a [falsy value](https://developer.mozilla.org/en-US/docs/Glossary/Falsy). For objects, this distinction typically doesn't matter, but it does for primitive types. For example, 0 or the empty string could be valid values, but are falsy (and thus treated as if they were invalid). Still, it's good to always be as intentional as possible and the semantics of `??` is preferable here.